app/src/main/java/com/android/streetworkapp/ui/park/AddEvent.kt

```diff
@@ -0,0 +1,305 @@
+            onValueChange = {
+              sliderPosition = it
+              event.maxParticipants = sliderPosition.toInt()
+            },
+            colors =
+                SliderDefaults.colors(
+                    thumbColor = MaterialTheme.colorScheme.secondary,
+                    activeTrackColor = MaterialTheme.colorScheme.secondary,
+                    inactiveTrackColor = MaterialTheme.colorScheme.secondaryContainer,
+                ),
+            steps = 7,
+            valueRange = minParticipants.toFloat()..maxParticipants.toFloat())
+        Text(text = sliderPosition.toInt().toString())
+      }
+}
+
+/**
+ * Used to change the date and the hour of the event
+ *
+ * @param event the event that will be updated
+ */
+@ExperimentalMaterial3Api
+@Composable
+fun TimeSelection(event: Event) {
+  var showDatePicker by remember { mutableStateOf(false) }
+  var showTimePicker by remember { mutableStateOf(false) }
+
+  val currentTime = Calendar.getInstance()
+  val datePickerState = rememberDatePickerState(currentTime.timeInMillis)
+  val timePickerState =
+      rememberTimePickerState(
+          initialHour = currentTime.get(Calendar.HOUR_OF_DAY),
+          initialMinute = currentTime.get(Calendar.MINUTE),
+          is24Hour = false,
+      )
+
+  val currentTimeSelection =
+      datePickerState.selectedDateMillis?.plus(
+          (TimeUnit.HOURS.toMillis(timePickerState.hour.toLong()) +
+              TimeUnit.MINUTES.toMillis(timePickerState.minute.toLong())))
+
+  val selectedDate =
+      datePickerState.selectedDateMillis?.let { convertMillisToDate(currentTimeSelection!!) } ?: ""
+
+  Box(modifier = Modifier.fillMaxWidth(0.9f)) {
+    OutlinedTextField(
+        value = selectedDate,
+        onValueChange = {},
+        label = { Text("When do you want to train?") },
+        readOnly = true,
+        trailingIcon = {
+          Row {
+            IconButton(
+                modifier = Modifier.testTag("dateIcon"),
+                onClick = { showDatePicker = !showDatePicker }) {
+                  Icon(imageVector = Icons.Default.DateRange, contentDescription = "Select date")
+                }
+            IconButton(
+                modifier = Modifier.testTag("timeIcon"),
+                onClick = { showTimePicker = !showTimePicker }) {
+                  Icon(imageVector = Icons.Default.AddCircle, contentDescription = "Select time")
+                }
+          }
+        },
+        modifier = Modifier.fillMaxWidth().height(64.dp))
+
+    if (showDatePicker) {
+      Popup(onDismissRequest = { showDatePicker = false }, alignment = Alignment.TopStart) {
+        Box(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .shadow(elevation = 4.dp)
+                    .background(MaterialTheme.colorScheme.surface)
+                    .padding(16.dp)) {
+              DatePicker(state = datePickerState, showModeToggle = false)
+            }
+
+        Button(
```

Comment:
Improvement: I can validate dates that are earlier in time (e.g October 2020), we'll probably do the checks in the viewmodel and whatnot anyways but it would be a good addition to the UI that a user can't validate an invalid date

Reply:
You are right, this should be improved. Unfortunately, this task is harder than it seems. Do you mind if we do this update in another future sprint?

Reply:
Yep let's do that